common/storage/memory/keyvalue_test.go

```diff
@@ -0,0 +1,130 @@
+package memory
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/pastelnetwork/gonode/common/storage"
+	"github.com/stretchr/testify/assert"
+)
+
+// newTestDB. need keep as private method to prevent CI error:
+// exported func NewTestDB returns unexported type *memory.keyValue, which can be annoying to use
+func newTestDB() *keyValue {
+	return &keyValue{
+		values: map[string][]byte{
+			"exist": []byte("bar"),
+		},
+	}
+}
+
+func TestGet(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		key           string
+		expectedError error
+		expectedValue []byte
+	}{
+		{
+			key:           "exist",
+			expectedError: nil,
+			expectedValue: []byte("bar"),
+		}, {
+			key:           "not-exist",
+			expectedError: storage.ErrKeyNotFound,
+			expectedValue: nil,
+		},
+	}
+
+	t.Run("group", func(t *testing.T) {
+		for _, testCase := range testCases {
+
+			testCase := testCase
+			testName := fmt.Sprintf("key:%s/value:%v/err:%v", testCase.key, testCase.expectedValue, testCase.expectedError)
+			t.Run(testName, func(t *testing.T) {
+				t.Parallel()
+				db := newTestDB()
+				val, err := db.Get(testCase.key)
+				assert.Equal(t, testCase.expectedError, err)
+				assert.Equal(t, testCase.expectedValue, val)
+			})
+		}
+
+	})
+}
+
+func TestSet(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		key                  string
+		value                []byte
+		expectedError        error
+		expectedLengthValues int
+	}{
+		{
+			key:                  "exist",
+			value:                []byte("baz"),
+			expectedError:        nil,
+			expectedLengthValues: 1,
+		}, {
+			key:                  "foo",
+			value:                []byte("grid"),
+			expectedError:        nil,
+			expectedLengthValues: 2,
+		},
+	}
+
+	t.Run("group", func(t *testing.T) {
+		for _, testCase := range testCases {
+
+			testCase := testCase
+			testName := fmt.Sprintf("key:%s/value:%v/length:%d/err:%v", testCase.key, testCase.value, testCase.expectedLengthValues, testCase.expectedError)
+			t.Run(testName, func(t *testing.T) {
+				t.Parallel()
+				db := newTestDB()
+				err := db.Set(testCase.key, testCase.value)
+				assert.Equal(t, testCase.expectedError, err)
+				value := db.values[testCase.key]
```

Comment:
Make sure, that key doesn't exist
```suggestion
				value, ok := db.values[testCase.key]
				assert.True(t, ok, "not found a new key")
```

Reply:
ok got it